templates/cdap-deployment.yaml

```diff
@@ -88,6 +88,33 @@ spec:
           {{if $c.ImagePullPolicy}}
           imagePullPolicy: {{$c.ImagePullPolicy}}
           {{end}}
+          {{if $c.Lifecycle}}
```

Comment:
Can we directly serialize the `Lifecycle` object instead of manually copying individual components? Otherwise you end up with a lot of nested if statements. For an example, please see https://github.com/cdapio/cdap-operator/pull/73, specifically [deployment.go#341](https://github.com/cdapio/cdap-operator/blob/fdf0de41674e41b537ffa186b006d2407a61a31a/controllers/deployment.go#L341) and [deployment.go#372](https://github.com/cdapio/cdap-operator/blob/fdf0de41674e41b537ffa186b006d2407a61a31a/controllers/deployment.go#L372).

Reply:
Thanks for the suggestion. Updated.